staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go

```diff
@@ -653,7 +653,9 @@ type CDICertConfig struct {
 type CDISpec struct {
 	// +kubebuilder:validation:Enum=Always;IfNotPresent;Never
 	// PullPolicy describes a policy for if/when to pull a container image
-	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty" valid:"required"`
+	ImagePullPolicy  corev1.PullPolicy             `json:"imagePullPolicy,omitempty" valid:"required"`
+	// The imagePullSecrets to pull the container images
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```

Comment:
I assume that we want regular users to know the names of secrets we expect them to create?  So I suggest moving `ImagePullSecrets` to `CDIConfig`.  So CDI will look like this:

```yaml
...
kind: CDI
spec:
  config:
    imagePullSecrets:
    - secret-name
...
```

And CDIConfig should have `imagePullSecrets` in the status like so:

```yaml
...
kind: CDIConfig
metadata:
  name: config
spec:
  imagePullSecrets:
    - secret-name
status:
  imagePullSecrets:
    - secret-name
...
```

CDIConfig has a singleton (named config) that is created by the config-controller and readable by all users.  The `spec` section is `spec.config` from the CDI resource.

I refer you again to how `PodResourceRequirements` are handled